From 655b07b701271bc00952fe64aeb14f993a48a50e Mon Sep 17 00:00:00 2001
From: Christian Marangi <ansuelsmth@gmail.com>
Date: Tue, 16 Apr 2024 16:17:36 +0200
Subject: [PATCH 5/6] nss_dp_main: use phy_detach instead of disconnect in
 dp_remove

Use phy_detach instead of disconnect in dp_remove. On Module remove, phy
are already disconnected but they need to be detached to be correctly
reattached later with an insmod.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
---
 nss_dp_main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/nss_dp_main.c
+++ b/nss_dp_main.c
@@ -981,7 +981,7 @@ static int nss_dp_remove(struct platform
 		unregister_netdev(dp_priv->netdev);
 
 		if (dp_priv->phydev)
-			phy_disconnect(dp_priv->phydev);
+			phy_detach(dp_priv->phydev);
 
 #if defined(NSS_DP_PPE_SUPPORT)
 		/*
